main.go

```diff
@@ -307,6 +306,15 @@ func main() {
 	svc.Logger.Info("LNDhub exiting gracefully. Goodbye.")
 }
 
+func createLoggingMiddleware(logger *lecho.Logger) echo.MiddlewareFunc {
+	return lecho.Middleware(lecho.Config{
+		Logger: logger,
+		Enricher: func(c echo.Context, logger zerolog.Context) zerolog.Context {
+			return logger.Interface("UserID", c.Get("UserID"))
```

Comment:
does this work if the UserID is not there or must the middleware only be added to authenticated endpoints?

Reply:
If the userID is not there it will just log `nil` . It is also added to unauthenticated endpoints, see below. We just don't add it on the landing page and on the swagger page.